tests/Meziantou.Analyzer.Test/Rules/AvoidComparisonWithBoolLiteralAnalyzerTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Threading.Tasks;
+using Meziantou.Analyzer.Rules;
+using TestHelper;
+using Xunit;
+
+namespace Meziantou.Analyzer.Test.Rules
+{
+    public sealed class AvoidComparisonWithBoolLiteralAnalyzerTests
```

Comment:
You should add more tests for valid cases

````
bool a = true;
if (a) // no diagnostic

if (!a) // no diagnostic
````

Reply:
Added a bit more test cases